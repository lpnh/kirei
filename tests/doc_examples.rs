use kirei::AskamaFormatter;

fn create_formatter() -> AskamaFormatter {
    AskamaFormatter::new().expect("Failed to create formatter")
}

#[test]
fn base_template_1() {
    let mut formatter = create_formatter();
    let input = r#"<!DOCTYPE html><html lang="en"><head><title>{% block title %}{{ title }} - My Site{% endblock %}</title>{% block head %}{% endblock %}</head><body><div id="content">{% block content %}<p>Placeholder content</p>{% endblock %}</div></body></html>"#;

    let result = formatter.format(input).expect("Formatting failed");

    insta::assert_snapshot!(result);
}

#[test]
fn for_1() {
    let mut formatter = create_formatter();
    let input =
        r#"<h1>Users</h1><ul>{% for user in users %}<li>{{ user.name }}</li>{% endfor %}</ul>"#;

    let result = formatter.format(input).expect("Formatting failed");

    insta::assert_snapshot!(result);
}

#[test]
fn comment_1() {
    let mut formatter = create_formatter();
    let input = r#"{# A Comment #}"#;

    let result = formatter.format(input).expect("Formatting failed");

    insta::assert_snapshot!(result);
}
