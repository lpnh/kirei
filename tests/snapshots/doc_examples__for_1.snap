---
source: tests/doc_examples.rs
expression: result
---
<h1>Users</h1>
<ul>
  {% for user in users %}
  <li>{{ user.name }}</li>
  {% endfor %}
</ul>
